core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -469,8 +474,8 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    def maybeFlushAndCommitOffsets() {
-      if (System.currentTimeMillis() - lastOffsetCommitMs > offsetCommitIntervalMs) {
+    def maybeFlushAndCommitOffsets(autoCommit: Boolean) {
+      if (!autoCommit || System.currentTimeMillis() - lastOffsetCommitMs > offsetCommitIntervalMs) {
```

Comment:
Not sure why we need to have the boolean flag here. It seems that the logic should be `if (mirrormakerConsumer.commitRequested() || System.currentTimeMillis() - lastOffsetCommitMs > offsetCommitIntervalMs)` right?

The current solution is guaranteed to wait for `consumer.timeout.ms` during the rebalance, given that we hard coded it to one second it is probably fine.